### PR TITLE
BD-21 feat: add start version of modal component

### PR DESCRIPTION
**Pull request by pRomKol**

### CODE REVIEW

File: src/components/modal/Modal.stories.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  component: Modal,
+  title: 'Modal',
+} satisfies Meta<typeof Modal>
+
+export default meta
+
+export const Default: StoryObj<typeof meta> = {
+  args: {
+    children: 'some text',
+    title: 'some title',
```
Comment (sfgardev) [inline src/components/modal/Modal.stories.ts:15]: image

File: src/components/modal/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    <div
+      className={cn(
+        'w-fit divide-y-[1px] divide-dark-100 rounded-sm border border-dark-100 bg-dark-300'
+      )}
+    >
+      <div className='flex flex-row justify-between px-[24px] pb-[11px] pt-[12px]'>
+        <h3 className='text-xl font-bold leading-[36px] text-light-100'>
+          {title}
+        </h3>
+        <Icon
```
Comment (sfgardev) [inline src/components/modal/Modal.tsx:31]: add cursor pointer to close icon

File: src/components/modal/Modal.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+          className='mt-[7px]'
+          color='white'
+          icon='close-outline'
+          onClick={() => setIsOpen(false)}
+        />
+      </div>
+      <div className={cn('flex flex-col px-[24px] text-white', className)}>
```
Comment (sfgardev) [inline src/components/modal/Modal.tsx:38]: add vertical padding

File: src/components/modal/Modal.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+export const Modal = ({
+  children,
+  className,
+  isOpened,
+  title,
+}: ModalProps & PropsWithChildren<{ className?: string }>) => {
+  const [isOpen, setIsOpen] = useState(isOpened)
+
+  return isOpen ? (
+    <div
```
Comment (sfgardev) [inline src/components/modal/Modal.tsx:22]: modal should have some max-width, should be in overlay

**Review by sfgardev — CHANGES_REQUESTED** [review]

**Comment by sfgardev** [comment]
and add index.ts reexport in modal folder and also export in src/index.ts

File: src/components/modal/Modal.stories.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { Meta, StoryObj } from '@storybook/react'
+
+import { Modal } from './Modal'
+
+const meta = {
+  component: Modal,
+  title: 'Modal',
```
[the remark this answers is not in the document]
Comment (pRomKol) [reply src/components/modal/Modal.stories.ts:7]: done

File: src/components/modal/Modal.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  isOpened: boolean
+  title: string
+} & ComponentProps<'div'>
+
+export const Modal = ({
+  children,
+  className,
+  isOpened,
+  title,
+}: ModalProps & PropsWithChildren<{ className?: string }>) => {
```
[the remark this answers is not in the document]
Comment (pRomKol) [reply src/components/modal/Modal.tsx:18]: done

File: src/components/modal/Modal.stories.ts (hunk 1 above)
Comment (pRomKol) [reply src/components/modal/Modal.stories.ts:15]: done

File: src/components/modal/Modal.tsx (hunk 2 above)
Comment (pRomKol) [reply src/components/modal/Modal.tsx:31]: done

File: src/components/modal/Modal.tsx (hunk 3 above)
Comment (pRomKol) [reply src/components/modal/Modal.tsx:38]: done

File: src/components/modal/Modal.tsx (hunk 4 above)
Comment (pRomKol) [reply src/components/modal/Modal.tsx:22]: done

**Review by sfgardev — CHANGES_REQUESTED** [review]

File: src/components/modal/Modal.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+  return (
+    <div
+      className='fixed inset-0 z-50 flex items-center justify-center bg-light-900 bg-opacity-60'
+      onClick={onClose}
+    >
+      <div
+        aria-labelledby='modal-title'
+        className={cn(
+          'w-fit max-w-[500px] divide-y-[1px] divide-dark-100 rounded-sm border border-dark-100 bg-dark-300'
```
[the remark this answers is not in the document]
Comment (pRomKol) [reply src/components/modal/Modal.tsx:46]: done

File: src/components/modal/Modal.tsx [hunk 8]
```diff
@@ -0,0 +1,77 @@
+import { ComponentProps, PropsWithChildren, useEffect } from 'react'
+
+import { cn } from '../../utils'
+import { Icon } from '../icon'
+
+export type ModalProps = {
+  className?: string
```
[the remark this answers is not in the document]
Comment (pRomKol) [reply src/components/modal/Modal.tsx:7]: done

File: src/components/modal/Modal.tsx [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+  title: string
+} & ComponentProps<'div'>
+
+export const Modal = ({
+  children,
+  className,
+  isOpened,
+  onClose,
+  title,
+}: ModalProps & PropsWithChildren<{ className?: string }>) => {
```
[the remark this answers is not in the document]
Comment (pRomKol) [reply src/components/modal/Modal.tsx:19]: done

File: src/components/modal/Modal.stories.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+
+import { Modal } from './Modal'
+
+const meta: Meta<typeof Modal> = {
+  component: Modal,
+}
+
+export default meta
+
+export const Default: StoryObj<typeof Modal> = {
```
[the remark this answers is not in the document]
Comment (pRomKol) [reply src/components/modal/Modal.stories.ts:11]: done

File: src/components/modal/index.ts [hunk 11]
```diff
@@ -0,0 +1 @@
+export { Modal } from './Modal'
```
[the remark this answers is not in the document]
Comment (pRomKol) [reply src/components/modal/index.ts:1]: done

File: src/components/modal/Modal.tsx [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+import { Icon } from '../icon'
+
+export type ModalProps = {
+  className?: string
+  isOpened: boolean
+  onClose: () => void
+  title: string
+} & ComponentProps<'div'>
+
+export const Modal = ({
```
[the remark this answers is not in the document]
Comment (pRomKol) [reply src/components/modal/Modal.tsx:13]: done

**Review by sfgardev — APPROVED** [review]

**Pull request merged**